File: exercice2_4.c
#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>

pid_t pid_fils;

int main ()
{
    int i;
    printf("je suis le père\n");
    for(i=0;i<5;i++)
    {
         pid_fils=fork();
        if(pid_fils==0)
        {
            printf("je suis le fils %d\n",i);
            break;
        }
    }
    return(0);
}

File: exercice2_4.py
import os

pid_fils = 0  

def main():
    global pid_fils

    print("je suis le père")

    for i in range(5):
        pid_fils = os.fork()

        if pid_fils == 0:
            print(f"je suis le fils {i}")
            break

main()